src/devices/Ft232H/Ft232HSpi.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.Linq;
+
+namespace Iot.Device.Ft232H
+{
+    /// <summary>
+    /// SPI Device for FT232H
+    /// </summary>
+    public class Ft232HSpi : SpiDevice
+    {
+        private readonly SpiConnectionSettings _settings;
+
+        /// <inheritdoc/>
+        public override SpiConnectionSettings ConnectionSettings => _settings;
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public Ft232HDevice DeviceInformation { get; internal set; }
+
+        internal Ft232HSpi(SpiConnectionSettings settings, Ft232HDevice deviceInformation)
+        {
+            DeviceInformation = deviceInformation;
+            _settings = settings;
+            if ((_settings.ChipSelectLine < 3) || (_settings.ChipSelectLine > Ft232HDevice.PinCountConst))
+            {
+                throw new ArgumentException($"Chip Select line has to be between 3 and {Ft232HDevice.PinCountConst - 1}");
+            }
+
+            if (DeviceInformation._connectionSettings.Where(m => m.ChipSelectLine == _settings.ChipSelectLine).Any())
+            {
+                throw new ArgumentException("Chip Select already in use");
+            }
+
+            // Open the device
+            DeviceInformation._connectionSettings.Add(_settings);
+            DeviceInformation.SpiInitialize();
+        }
+
+        /// <inheritdoc/>
+        public override void Read(Span<byte> buffer)
+        {
+            DeviceInformation.SpiRead(_settings, buffer);
+        }
+
+        /// <inheritdoc/>
+        public override byte ReadByte()
+        {
+            Span<byte> buffer = stackalloc byte[1];
+            DeviceInformation.SpiRead(_settings, buffer);
+            return buffer[0];
+        }
+
+        /// <inheritdoc/>
+        public override void TransferFullDuplex(ReadOnlySpan<byte> writeBuffer, Span<byte> readBuffer)
+        {
+            DeviceInformation.SpiWriteRead(_settings, writeBuffer, readBuffer);
+        }
+
+        /// <inheritdoc/>
+        public override void Write(ReadOnlySpan<byte> buffer)
+        {
+            DeviceInformation.SpiWrite(_settings, buffer);
+        }
+
+        /// <inheritdoc/>
+        public override void WriteByte(byte value)
+        {
+            DeviceInformation.SpiWrite(_settings, stackalloc byte[1] { value });
```

Comment:
same comment as for I2C for reading/writing single byte

Reply:
uhh I just noticed that's actually abstract and we don't provide default implementation which I think we should... Not sure if that would be considered breaking change....

Reply:
I just started a conversation if it would be safe to make it virtual - I think many of our internal implementations do it exactly this way